### PR TITLE
git-credential-oauth 0.7.0 (new formula)

### LINKED ISSUE #22: Homebrew package
Hi,

I don't at the moment have a system with which to finish this, but here's a sample Homebrew Formulae that might work:

```ruby
# Documentation: https://docs.brew.sh/Formula-Cookbook
#                https://rubydoc.brew.sh/Formula
class GitCredentialOauth < Formula
  desc "A Git credential helper that securely authenticates to GitHub, GitLab and BitBucket using OAuth."
  homepage "https://github.com/hickford/git-credential-oauth"
  url "https://github.com/hickford/git-credential-oauth/archive/refs/tags/v0.7.0.tar.gz"
  sha256 "017bd47edc0dd3057323d8b9ccca008b7ebca7aedf6862b1ebca5e54f5a62496"
  #head "https://github.com/hickford/git-credential-oauth"
  license "Apache-2.0"

  livecheck do
    url :stable
    strategy :github_latest
  end

  depends_on "go" => :build

  def install
    system "go", "build", *std_go_args(ldflags: "-s -w", output: bin/"git-credential-oauth"), "main.go"
  end

  test do
    system bin/"git-credential-oauth", "configure"
  end

end
```

I don't know how you generate your builds (does Go just build for any architecture regardless of your host CPU?) but if you needed a Mac platform to test Homebrew on, CircleCI supports all major platforms and is free for OSS projects: https://circleci.com/open-source/

### PR DESCRIPTION
**Pull request by hickford**
https://github.com/hickford/git-credential-oauth

Fixes https://github.com/hickford/git-credential-oauth/issues/22

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/git-credential-oauth.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  head "https://github.com/hickford/git-credential-oauth.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w")
+  end
+
+  test do
+    shell_output("#{bin}/git-credential-oauth", 2)
```
Comment (chenrui333) [inline Formula/git-credential-oauth.rb:16]: need to assert the test

Comment (hickford) [reply Formula/git-credential-oauth.rb:16]: shell_output calls assert_equal with the given exit status . Am I missing something? https://rubydoc.brew.sh/Homebrew/Assertions.html#shell_output-instance_method

```ruby
def shell_output(cmd, result = 0)
  ohai cmd
  output = `#{cmd}`
  assert_equal result, $CHILD_STATUS.exitstatus
  output
rescue Minitest::Assertion
  puts output if verbose?
  raise
end
```

Comment (carlocab) [inline Formula/git-credential-oauth.rb:16]:
```suggestion
    refute_empty shell_output("#{bin}/git-credential-oauth", 2)
```
or
```suggestion
    assert_empty shell_output("#{bin}/git-credential-oauth", 2)
```
whichever one is the case

**Review by carlocab — APPROVED** [review]
Thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by hickford** [comment]
@carlocab @chenrui333 Might you be interested in packaging sister project https://github.com/hickford/git-credential-azure ?

**Pull request merged**